qiskit/quantum_info/synthesis/two_qubit_decompose.py

```diff
@@ -547,6 +544,73 @@ def specialize(self):
         self.K2r = np.asarray(RYGate(k2rtheta)) @ np.asarray(RXGate(k2rlambda))
 
 
+class TwoQubitWeylEchoRZX(TwoQubitWeylDecomposition):
+    """Decompose two-qubit unitary in terms of echoed cross-resonance gates."""
+
+    def __init__(self, unitary, inst_map, qubit_pair: Tuple):
+        self.inst_map = inst_map
+        self.qubit_pair = qubit_pair
+        super().__init__(unitary)
+
+    def specialize(self):
+        pass  # Nothing to do
+
+    @staticmethod
+    def _apply_rzx(circ: QuantumCircuit, angle: float):
+        """Echoed RZX gate"""
+        circ.rzx(-angle, 0, 1)
+        circ.x(0)
+        circ.rzx(angle, 0, 1)
+        circ.x(0)
+
+    @staticmethod
+    def _apply_reverse_rzx(circ: QuantumCircuit, angle: float):
+        """Reverse direction of the echoed RZX gate"""
+        circ.h(0)
+        circ.h(1)
+        circ.rzx(-angle, 1, 0)
+        circ.x(1)
+        circ.rzx(angle, 1, 0)
+        circ.x(1)
+        circ.h(0)
+        circ.h(1)
+
+    def is_native_cx(self, qubit_pair: Tuple) -> bool:
+        """Check that a CX for a qubit pair is native."""
+        cx1 = self.inst_map.get("cx", qubit_pair)
+        cx2 = self.inst_map.get("cx", qubit_pair[::-1])
+        return cx1.duration < cx2.duration
```

Comment:
Perhaps we can move this to `EchoRZXWeylDecomposition`. This would mean that we only need to provide `TwoQubitWeylEchoRZX` with the native direction instead of giving it two arguments that we use to figure out the direction.
```suggestion
```